upgrade-2.x.sh

```diff
@@ -483,56 +656,40 @@ else
     log "No supervisor updater fix is required..."
 fi
 
-# Find which partition is / and which we should write the update to
-root_part=$(findmnt -n --raw --evaluate --output=source /)
-log "Found root at ${root_part}..."
-case $root_part in
-    *2)
-        # on 2.x the following device types have these kinds of results for $root_part
-        # raspberrypi: /dev/mmcblk0p2
-        # beaglebone: /dev/disk/by-partuuid/93956da0-02
-        # NUC: /dev/sda2
-        root_dev=${root_part%2}
-        update_part=${root_dev}3
-        update_part_no=3
-        update_label=resin-rootB
-        ;;
-    *3)
-        root_dev=${root_part%3}
-        update_part=${root_dev}2
-        update_part_no=2
-        update_label=resin-rootA
-        ;;
-    *rootA)
-        old_label=resin-rootA
-        update_label=resin-rootB
-        root_part_dev=$(blkid | grep "${old_label}" | awk '{print $1}' | sed 's/://')
-        update_part=${root_part_dev%2}3
-        ;;
-    *rootB)
-        old_label=resin-rootB
-        update_label=resin-rootA
-        root_part_dev=$(blkid | grep "${old_label}" | awk '{print $1}' | sed 's/://')
-        update_part=${root_part_dev%2}3
-        ;;
-    *)
-        log ERROR "Unknown root partition \"$root_part\"."
-esac
-if [ ! -b "$update_part" ]; then
-    log ERROR "Update partition detected as ${update_part} but it's not a block device."
+### hostapp-update based updater
+
+if [ -x "$(command -v hostapp-update)" ]; then
```

Comment:
Why are we not using the OS version for this? `minimum_hostapp_target_version` is set at the top of the script

Reply:
I fell this is more robust to check whether the current system has hostapp-update support or not.. The `minimum_hostapp_target_version` is checking the update target, while this check is checking the current system. Can update to use the version check too, then.